core/dbt/events/functions.py

```diff
@@ -138,25 +137,21 @@ def event_to_serializable_dict(
     ts_fn: Callable[[datetime], str]
 ) -> Dict[str, Any]:
     data = dict()
-    node_info = dict()
+    node_info: Dict[str, Any] = dict()
     log_line = dict()
     try:
-        log_line = dataclasses.asdict(e, dict_factory=type(e).asdict)
-    except AttributeError:
+        log_line = e.to_dict()
+    except AttributeError as exc:
         event_type = type(e).__name__
         raise Exception(  # TODO this may hang async threads
-            f"type {event_type} is not serializable to json."
-            f" First make sure that the call sites for {event_type} match the type hints"
-            f" and if they do, you can override the dataclass method `asdict` in {event_type} in"
-            " types.py to define your own serialization function to a dictionary of valid json"
-            " types"
+            f"type {event_type} is not serializable. {str(exc)}"
         )
 
     if isinstance(e, NodeInfo):
-        node_info = dataclasses.asdict(e.get_node_info())
+        node_info = e.node_info
 
     for field, value in log_line.items():  # type: ignore[attr-defined]
-        if field not in ["code", "report_node_data"]:
+        if field not in ["code", "node_info", "log_version", "ts", "ts_rfc3339", "pid"]:
```

Comment:
should we generate this list at runtime instead? If we add a new field on `Event` we'd have to remember to put it here too.

Reply:
We don't actually have any fields on Event now. I've changed this to delete 'code' and 'node_info' from the dictionary, but I'm wondering if it wouldn't be better to just leave node_info in the data and not move it up.